### PR TITLE
Add configuration options for chat history plugin

### PR DESCRIPTION
**Pull request by brandongregoryscott**
This is my first time contributing, but I'd love to keep adding features! This pull request fixes issue #2418 by adding a configuration menu for the Chat History plugin. I'm not 100% sure of every situation where each of the `ChatMessageType` enums are fired, so I'd love some insight on that.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java [hunk 1]
```diff
@@ -109,12 +122,114 @@ public void onMenuOptionClicked(MenuOptionClicked event)
...
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("chathistory"))
+		{
+			populateAllowedMessages();
+
+			// If the event's new value is false, it means we need to check the message queue
+			// for any unwanted message types and remove them before the next hop or login
+			if (event.getNewValue().equals("false"))
+				purgeUnwantedMessages(event.getKey());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java]: I would rather just not replay messages based on config than try to filter the message queue based on the config.

File: runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java [hunk 2]
```diff
@@ -88,18 +98,21 @@ public void onSetMessage(SetMessage message)
...
-				.runeLiteFormattedMessage(nbsp(message.getMessageNode().getRuneLiteFormatMessage()))
-				.build();
+				.value(nbsp(message.getValue()));
 
-			if (!messageQueue.contains(queuedMessage))
-			{
-				messageQueue.offer(queuedMessage);
-			}
+			// Check to see if this is a formatted message from RuneLite - such the GE/HA value on examine
+			if (message.getValue().contains("<col"))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java]: why was this added?

**Comment by deathbeam** [comment]
The solution for this would be just not filter game chat and not add 90 toggles

File: runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java (hunk 2 above)
Comment (brandongregoryscott) [reply runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java]: This was added to ensure that colored messages (ie the extra message with the Grand Exchange & High Alch value when you examine an item) shows up colored when you hop or login. It wasn't maintaining the color style with the previous way the builder was setup. 

I'm not sure those should even be saved to the history, but they were showing up as the same chat message type, EXAMINE_ITEM, and showing up as all black text upon hopping, which looked off.

**Comment by brandongregoryscott** [comment]
@deathbeam The "GAME" message type is already allowed through the filter, though, but there's other message types that go through the game tab (server messages, examine messages, "filtered" messages) that are not saved. I personally wouldn't want most of these messages to be saved in the chat history without an option to turn them off, but I can remove or condense the configuration if that's what you think would be best.

File: runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryConfig.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	}
+
+	@ConfigItem(
+		keyName = "saveGameMessages",
+		name = "Save game messages",
+		description = "Configures whether or not to save history for messages from the game tab (does not include filtered ones)"
+	)
+	default boolean saveGameMessages()
+	{
+		return false;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryConfig.java]: I would set the configuration default values to be what the existing behavior is.

File: runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java (hunk 2 above)
Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java]: I don't really understand this. The RuneLite formatted message is not in the same format as the normal messages, so I don't think this would work right.

**Comment by Hydrox6** [comment]
Just a heads up, both RuneLite formatted and actual chat messages contain `<col`, as the way to colour chat messages is to add a `<col={hexcolour}>` before the message
